docs/reference/workflows/ecs/interaction/authority.md

```diff
@@ -2,114 +2,112 @@
+Below is an example of how to write a system that runs when a worker instance has authority over the `Position` SpatialOS component. You can iterate through the matching components using the ECS `.ForEach` syntax.
 
 ```csharp
 public class AuthoritativePositionSystem : ComponentSystem
 {
-    public struct Data
+    private EntityQuery query;
+
+    protected override void OnCreate()
     {
-        public readonly int Length;
-        // ECS component representing the SpatialOS component Position.
-        public ComponentDataArray<Position.Component> Position;
-        // ECS component tag representing the worker's authority over the SpatialOS Position component.
-        public ComponentDataArray<Authoritative<Position.Component>> PositionAuthority;
-    }
+        base.OnCreate();
 
-    [Inject] Data data;
+        query = GetEntityQuery(
+            ComponentType.ReadWrite<Position.Component>(),
+            ComponentType.ReadOnly<Position.ComponentAuthority>()
+        );
+
+        // Position.ComponentAuthority.Authoritative is a static copy of the
+        // Position.ComponentAuthority component with HasAuthority set to true
+        query.SetFilter(Position.ComponentAuthority.Authoritative);
+    }
 
     protected override void OnUpdate()
     {
-        for(var i = 0; i < data.Length; i++)
+        Entities.With(query).ForEach((ref Position.Component position) =>
         {
-            // This will only run when the worker instance is authoritative over a Position component.
-        }
+            // This will only run for entities for which the worker instance
+            // is authoritative over the Position component.
+        });
     }
 }
 ```
 
-## What happens when a SpatialOS entity enters a worker instance's view
-
-> This section is just to tell you how the system works: all of this is handled automatically by the GDK, and you don't need to do anything to ensure that authority is correctly registered in authority tags.
+## How to interact with authority changes
 
-When a SpatialOS entity enters the [worker instance's view]({{urlRoot}}/reference/glossary#worker-s-view), the GDK:
+The `GetAuthorityChangesReceived` method on the [`ComponentUpdateSystem`]({{urlRoot}}/api/core/component-update-system) allows you to retrieve a list of all the authority changes for an entity-component pair that have occured since the previous frame.
 
-- creates an ECS entity to correspond to that SpatialOS entity
-- for each SpatialOS component `T` on the SpatialOS entity:
-    - attaches an ECS component to the ECS entity representing the SpatialOS component `T`
-    - attaches a [`NotAuthoritative<T>`]({{urlRoot}}/api/reactive-components/not-authoritative) authority tag to represent the worker instance's authority over the SpatialOS component `T`
+> The list of authority changes retreived by `GetAuthorityChangesReceived` will indicate whether your authority state is either `NotAuthoritative`, `Authoritative` or `AuthorityLossImminent`.
```

Comment:
```suggestion
> The list of authority changes retrieved by `GetAuthorityChangesReceived` indicates whether your authority state is `NotAuthoritative`, `Authoritative` or `AuthorityLossImminent`.
```

Reply:
this bit has been moved to a separate section at the bottom of the page